src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs

```diff
@@ -131,8 +131,19 @@ PEFile MakePEFile(IAssemblySymbol assembly)
                 // reference assemblies should be fine here, we only need the metadata of references.
                 var reference = _parentCompilation.GetMetadataReference(assembly);
                 Log(CSharpEditorResources.Load_from_0, reference.Display);
-                return TryResolve(reference, PEStreamOptions.PrefetchMetadata)
-                    ?? new PEFile(reference.Display, PEStreamOptions.PrefetchMetadata);
+
+                var result = TryResolve(reference, PEStreamOptions.PrefetchMetadata);
+                if (result is not null)
+                {
+                    return result;
+                }
+
+                if (File.Exists(reference.Display))
+                {
+                    return new PEFile(reference.Display, PEStreamOptions.PrefetchMetadata);
+                }
+
+                return null;
```

Comment:
Since this path used to return non-null, do we guarantee this `null` can't flow somewhere and dereferenced causing NRE?

Reply:
The containing method returns null for another case (line 82) so I imagine its fine.